src/org/zalando/stups/friboo/system/http.clj

```diff
@@ -172,6 +181,88 @@
     (when (:join? options true) (.join server))
     server))
 
+(defn add-logs
+  "Adds a list of log entries to the audit logs container."
+  [audit-logs logs]
+  (dosync
+    (alter audit-logs concat logs)))
+
+(defn empty-logs
+  "Returns all current audit logs and resets the container to empty."
+  [audit-logs]
+  (dosync
+    (let [previous-logs @audit-logs]
+      (ref-set audit-logs [])
+      previous-logs)))
+
+(defn is-modifying? [{:keys [request-method]}]
+  (some #(= % request-method) [:post :put :patch :delete]))
+
+(defn collect-audit-logs
+  "Adds request map to audit-logs container."
+  [next-handler audit-logs enabled?]
+  (if enabled?
+    (do
+      (log/info "Logging all successful, modifying requests.")
+      (fn [request]
+        (let [{:keys [status] :as response} (next-handler request)]
+          (when (and (>= status 200)
+                     (< status 300)
+                     (is-modifying? request))
+            (let [audit-log (-> request
+                                (assoc :logged-on (t/now))
+                                (dissoc :swagger)
+                                (dissoc :configuration)
+                                (dissoc :body)
+                                (dissoc-in [:tokeninfo "access_token"]))]
```

Comment:
Would it make sense, instead of excluding some fields, just to explicitly include
* URI
* HTTP method
* parsed params?